packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:file/memory.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/tools/asset_transformer.dart';
+import 'package:flutter_tools/src/flutter_manifest.dart';
+
+import '../../../src/common.dart';
+import '../../../src/fake_process_manager.dart';
+
+void main() {
+  test('Invokes dart properly', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')..createSync()..writeAsStringSync('hello world');
+    const String outputPath = 'output.txt';
+
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          artifacts.getArtifactPath(Artifact.engineDartBinary),
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+          '-f',
+          '--my_option',
+          'my_option_value',
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+            ..addOption('input')
+            ..addOption('output')
+            ..addFlag('foo', abbr: 'f')
+            ..addOption('my_option'))
+            .parse(args);
+          fileSystem.file(parsedArgs['output']).createSync(recursive: true);
+        },
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: artifacts.getArtifactPath(Artifact.engineDartBinary),
+    );
+
+    await transformer.transformAsset(
+      asset: asset,
+      outputPath: outputPath,
+      workingDirectory: fileSystem.currentDirectory.path,
+      transformerEntries: <AssetTransformerEntry>[
+        const AssetTransformerEntry(
+          package: 'my_transformer',
+          args: <String>[
+            '-f',
+            '--my_option',
+            'my_option_value',
+          ],
+        )
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+  });
+
+  testWithoutContext('logs useful error information when transformation process returns a nonzero exit code', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
```

Comment:
`MemoryFileSystem.test()`

Reply:
Addressed in [nit pack 1](https://github.com/flutter/flutter/pull/143815/commits/898a4b6396aee4098bce759b35caec356c96809a)